### PR TITLE
WIP: Set up work-in-progress github actions workflows

### PR DESCRIPTION
**Pull request by francislavoie**
This may or may not be nicer than Azure Pipelines (same backend, different UI, slightly different config)

### CODE REVIEW

**Comment by mholt** [comment]
Thanks Francis! Should I enable GitHub actions for this repo to test this?

**Comment by francislavoie** [comment]
That would be great, thanks.

Still a lot of open questions, hopefully @Mohammed90 can help me here 😄

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      id: step_test
+      continue-on-error: true
+      run: |
+        (go test -v -coverprofile=cover-profile.out -race ./... 2>&1) > test-results/test-result.out
+        echo "::set-output name=status::$?"
+
+    - name: Run integration tests
+      run: |
+        set -e
+        cmd/caddy/caddy start
```
Comment (sarge) [inline .github/workflows/ci.yml]: Thanks for including this step, I wasn't completely finished with the windows version. I think you will need to do something with the path separators, I am not sure it set -e is supported either.

Comment (francislavoie) [reply .github/workflows/ci.yml]: Okay thanks for the heads up, will probably leave it commented out until you give the 👍. Might just set it to not run on windows

**Comment by mholt** [comment]
Okay, it appears that the action will be triggered by simply committing the workflow files.

**Comment by francislavoie** [comment]
Hmm, I'm thinking the branch needs to be on upstream trigger it - I was having permission issues last night with pushing a new branch from GitKraken. I'll need to try some things.

**Pull request closed without merging**